src/e2e/create_pool.rs

```diff
@@ -168,4 +177,97 @@ pub mod e2e_tests {
 
         Ok(())
     }
+
+    #[ink_e2e::test]
+    async fn test_create_pool_init_sqrt_price_minimal_difference_from_tick(
+        mut client: ink_e2e::Client<C, E>,
+    ) -> E2EResult<()> {
+        let dex = create_dex!(client, ContractRef, Percentage::new(0));
+        let (token_x, token_y) = create_tokens!(client, TokenRef, 500, 500);
+        let alice = ink_e2e::alice();
+
+        let fee_tier = FeeTier::new(Percentage::from_scale(5, 1), 3).unwrap();
+        let init_tick = 0;
+        let init_sqrt_price = calculate_sqrt_price(init_tick).unwrap() - SqrtPrice::new(1);
```

Comment:
The sqrt price is now `0.99999999...` which means that the tick should be `0 - tick_spacing`. Probably validation is incorrect.